include/albatross/src/details/traits.hpp

```diff
@@ -161,6 +161,35 @@ template <typename... Ts>
 struct variant_size<variant<Ts...>>
     : public std::tuple_size<std::tuple<Ts...>> {};
 
+/*
+ * Eigen helpers
+ */
+template <typename T> class is_eigen_plain_object {
+  template <typename C>
+  static typename std::enable_if<
+      std::is_base_of<Eigen::PlainObjectBase<C>, C>::value,
+      std::true_type>::type
+  test(int);
+
+  template <typename C> static std::false_type test(...);
+
+public:
+  static constexpr bool value = decltype(test<T>(0))::value;
+};
+
+template <typename T> class is_eigen_xpr {
+  template <
+      typename C,
+      typename BaseType = typename Eigen::internal::dense_xpr_base<C>::type,
```

Comment:
Is the `BaseType` parameter needed? I'm not seeing how it's used.

Reply:
Yeah it actually is used, but I should make it more clear how.  Basically `Eigen::internal::dense_xpr_base<C>::type` will fail if `C` isn't an eigen matrix like type, so that line checks that it's possibly an eigen type, and the next line checks if it isn't a plain object.